packages/datepicker/datepicker.scss

```diff
@@ -21,6 +42,121 @@ $bottom-margin: $component-spacing--large;
+        border: none;
+        outline: none;
+        border-radius: 0;
+        color: inherit;
+
+        box-sizing: border-box;
+        width: 100%;
+        height: $input-height;
+        padding: 0 0 $bottom-padding;
+
+        font-weight: $bold-weight;
+        background-color: transparent;
+        transition: box-shadow $transition-timing;
+        @include underline-color(currentColor, $border-color--focus);
+        /* show border in mobile Safari */
+        -webkit-appearance: none;
+        resize: none; // Disable textarea resize
+
+        &::placeholder {
+            @include jkl-text-style("desktop/body");
+            font-weight: normal;
+            opacity: 1;
+            color: $gra-60;
+        }
+
+        &:hover,
+        &:focus {
+            box-shadow: inset 0 rem(-4px) 0 0 currentColor;
+        }
+
+        &[type="number"] {
+            text-align: right;
+        }
+
+        &[aria-invalid="true"],
+        &:invalid {
+            box-shadow: inset 0 rem(-2px) 0 0 $error-color;
+
+            &:focus,
+            &:hover,
+            html:not([data-mousenavigation]) &:focus {
+                box-shadow: inset 0 rem(-4px) 0 0 $error-color;
+            }
+        }
+    }
+
+    &__action-button {
+        position: absolute;
+        right: 0;
+        display: flex;
+        align-items: center;
+        top: 0;
+        appearance: none;
+        border: none;
+        background-color: transparent;
+        cursor: pointer;
+        padding: 0;
+        padding-bottom: $component-spacing--small;
+        height: $input-height;
+        color: $svart;
+
+        *:not([data-mousenavigation]) &:focus {
+            color: $focus-color;
+        }
+    }
+
+    &__action-icon {
+        display: inline-block;
+        width: $icon-size;
+    }
+
+
+
+    &--action {
+        position: relative;
+
+        .jkl-datepicker-text-input {
+            padding-right: $component-spacing--small + $icon-size;
+        }
+    }
+
+    @include compact-mode {
+        & .jkl-datepicker-text-input,
+        & .jkl-datepicker-text-input::placeholder {
+            @include jkl-text-style("compact/body");
+        }
+        & .jkl-datepicker-text-input {
+            height: $input-height--small-device;
+        }
+        & .jkl-datepicker__action-button {
+            height: $input-height--small-device;
+        }
+        & .jkl-datepicker__action-icon {
+            width: $icon-size--compact;
+        }
+    }
+
+    @include small-device {
+        .jkl-datepicker-text-input,
+        .jkl-datepicker-text-input::placeholder {
+            @include jkl-text-style("compact/body");
+        }
+        & .jkl-datepicker-text-input {
+            height: $input-height--small-device;
+        }
+        & .jkl-datepicker__action-button {
+            height: $input-height--small-device;
+        }
+        & .jkl-datepicker__action-icon {
+            width: $icon-size--compact;
+        }
+    }
```

Comment:
These should probably be moved to the bottom of the `.jkl-datepicker` class, otherwise their rules might be accidentally overridden by other properties in the class. The contents of the `@inlcude small-device` on line 350 can be merged into this one.